scheduler/testing.go

```diff
@@ -130,20 +127,47 @@ func (h *Harness) SubmitPlan(plan *structs.Plan) (*structs.PlanResult, State, er
 	// Setup the update request
 	req := structs.ApplyPlanResultsRequest{
 		AllocUpdateRequest: structs.AllocUpdateRequest{
-			Job:   plan.Job,
-			Alloc: allocs,
+			Job: plan.Job,
 		},
 		Deployment:        plan.Deployment,
 		DeploymentUpdates: plan.DeploymentUpdates,
 		EvalID:            plan.EvalID,
 		NodePreemptions:   preemptedAllocs,
 	}
 
+	if h.optimizePlan {
```

Comment:
I didn't find any usages for the SubmitPlan method of Harness, but changed the code to be able to support the newer format of the struct.